vault/data_source_azure_access_credentials.go

```diff
@@ -238,49 +238,52 @@ func azureAccessCredentialsDataSourceRead(ctx context.Context, d *schema.Resourc
 		return diag.Errorf("failed to create providers client: %s", err)
 	}
 	delay := time.Duration(d.Get("num_seconds_between_tests").(int)) * time.Second
-	endTime := time.Now().Add(
-		time.Duration(d.Get("max_cred_validation_seconds").(int)) * time.Second)
+	maxValidationDuration := d.Get("max_cred_validation_seconds").(int)
```

Comment:
nit: if this is named `*Duration`, I think it should be of type `time.Duration`.
```suggestion
	maxValidationDuration := time.Duration(d.Get("max_cred_validation_seconds").(int)) * time.Second
```